src/main/java/bubbles/tasks/Deadline.java

```diff
@@ -0,0 +1,69 @@
+package bubbles.tasks;
+
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+
+/**
+ * A child class of Task, representing the tasks that need to be
+ * done before a specific date/time.
+ */
+public class Deadline extends Task {
+    private LocalDate dueDate;
+
+    private Deadline(String description, boolean isDone, String dueDate) {
+        super(description, isDone);
+        this.dueDate = Task.formatDate(dueDate);
+    }
+
+    /**
+     * Act as a public constructor of a Deadline Object.
+     *
+     * @param input The description of the Deadline, including the due date
+     * @param isDone Whether the Deadline is done/completed
+     * @return The created Deadline Object
+     */
+    public static Deadline addDeadline(String input, boolean isDone) {
+        String[] arr = input.split(" /by ");
+
+        Deadline item = new Deadline(arr[0], isDone, arr[1]);
+
+        return item;
+    }
+
+    /**
+     * Format the Deadline Object to store in the hard disk File.
+     *
+     * @return String representing the Deadline Object (different from the String representation
+     *          of the Deadline Task)
+     */
+    @Override
+    public String format() {
+        String format = "D | ";
+
+        if (this.isDone) {
+            format += "1 | ";
+        } else {
+            format += "0 | ";
+        }
```

Comment:
What do you think about having Task do this (format + super.format() + ...)? Since this seems repeated across the 3 types of task and the format() method in Task seems to have the same functionality as its toString() method.